Install localized IntelliSense (#16234)
dotnet/docs: docs/core/install/localized-intellisense.md

---
title: Install localized IntelliSense files
description: Learn how to set up your development machine to use localized IntelliSense files for .NET Core projects in Visual Studio.
author: mairaw
ms.author: mairaw
ms.date: 12/18/2019
---
# How to install localized IntelliSense files for .NET Core

> IntelliSense is a code-completion aid that is available in different integrated development environments (IDEs), such as Visual Studio. By default, when you're developing .NET Core projects, the SDK only includes the English version of the IntelliSense files. This article explains:

Reviewer: ```suggestion
[IntelliSense](/visualstudio/ide/using-intellisense) is a code-completion aid that is available in different integrated development environments (IDEs), such as Visual Studio. By default, when you're developing .NET Core projects, the SDK only includes the English version of the IntelliSense files. This article explains:
```

- How to install the localized version of those files.

> - Modify the Visual Studio installation to use a different language.

Reviewer: ```suggestion
- How to modify the Visual Studio installation to use a different language.
```

## Prerequisites

> - [.NET Core 3.0 SDK](https://dotnet.microsoft.com/download/dotnet-core) or later versions.

Reviewer: ```suggestion
- [.NET Core 3.0 SDK](https://dotnet.microsoft.com/download/dotnet-core) or a later version.
```

> - [Visual Studio 2019 16.3](https://visualstudio.microsoft.com/downloads/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link&utm_content=download+vs2019) or later versions.

Reviewer: ```suggestion
- [Visual Studio 2019 version 16.3](https://visualstudio.microsoft.com/downloads/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link&utm_content=download+vs2019) or a later version.
```

## Download and install the localized IntelliSense files

> [!IMPORTANT]
> This procedure requires that you have administrator permission to copy the IntelliSense files to the .NET Core installation folder.

1. Go to the [Download IntelliSense files](https://dotnet.microsoft.com/download/dotnet-core) page.

1. Download the IntelliSense file for the language and version you'd like to use.

1. Extract the contents of the zip file.

> 1. Navigate to the .NET Core installation folder. By default, it's under *C:\Program Files\dotnet\packs*.

Reviewer: ```suggestion
1. Navigate to the .NET Core installation folder. By default, it's under *%ProgramFiles%\dotnet\packs*.
```
C: might not be their main drive letter.

>    - Choose which SDK you want to install the IntelliSense for and navigate to the associated path. You have the following options:

Reviewer: ```suggestion
   - Choose which SDK you want to install the IntelliSense for, and navigate to the associated path. You have the following options:
```

   - Open the *ref* folder.
   - Open the moniker folder. For example, *netcoreapp3.1*.

   So, the full path that you'd navigate to would look similar to *C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\3.1.0\ref\netcoreapp3.1*.

1. Create a subfolder inside the moniker folder you just opened. The name of the folder indicates which language you want to use. The following table specifies the different options:

   | Language              | Folder name |
   | --------------------- | ----------- |
   | Chinese (simplified)  | *zh-hans*   |
   | Chinese (traditional) | *zh-hant*   |
   | French                | *fr*        |
   | German                | *de*        |
   | Italian               | *it*        |
   | Japanese              | *ja*        |
   | Korean                | *ko*        |
   | Brazilian Portuguese  | *pt-br*     |
   | Russian               | *ru*        |
   | Spanish               | *es*        |

1. Copy the *.xml* files you extracted on step 3 to this new folder. The *.xml* files are broken down by SDK folders, so copy them to the matching SDK you chose on step 4.

## Modify Visual Studio language

> For Visual Studio to use a different language for IntelliSense, it requires you to install the language pack you want to use. This can be done [during installation](/visualstudio/install/install-visual-studio#step-6---install-language-packs-optional) or at a later time by modifying the Visual Studio installation. If you already have Visual Studio configured to the language of your choice, your IntelliSense installation is ready.

Reviewer: ```suggestion
For Visual Studio to use a different language for IntelliSense, install the appropriate language pack. This can be done [during installation](/visualstudio/install/install-visual-studio#step-6---install-language-packs-optional) or at a later time by modifying the Visual Studio installation. If you already have Visual Studio configured to the language of your choice, your IntelliSense installation is ready.
```

### Install the language pack

> If you haven't installed the desired language pack during installation, update your installation as follows to get the language pack installed:

Reviewer: ```suggestion
If you didn't install the desired language pack during setup, update Visual Studio as follows to install the language pack:
```

> [!IMPORTANT]
> To install, update, or modify Visual Studio, you must log on with an account that has administrative permissions. For more information, see [User permissions and Visual Studio](/visualstudio/ide/user-permissions-and-visual-studio).

1. Find the Visual Studio Installer on your computer.

   For example, on a computer running Windows 10, select **Start**, and then scroll to the letter **V**, where it's listed as **Visual Studio Installer**.

   ![Open the Visual Studio Installer from Windows](./media/localized-intellisense/vs-installer-windows-start.png)

   > [!NOTE]
   > You can also find the Visual Studio Installer in the following location:
   >
   > `C:\Program Files (x86)\Microsoft Visual Studio\Installer\vs_installer.exe`

   You might have to update the installer before continuing. If so, follow the prompts.

> 1. In the installer, look for the edition of Visual Studio that you installed, and then choose **Modify**.

Reviewer: ```suggestion
1. In the installer, look for the edition of Visual Studio that you want to add the language pack to, and then choose **Modify**.
```

   ![Update or modify Visual Studio](./media/localized-intellisense/vs-installer-modify.png)

   > [!IMPORTANT]
   > If you don't see a **Modify** button but you see an **Update** one instead, you need to update your Visual Studio before you can modify your installation.
   > Choose **Update**. After the update is finished, the **Modify** button should appear.

1. In the **Language packs** tab, select or deselect the languages you want to install or uninstall.

   ![Visual Studio language packs tab](./media/localized-intellisense/vs-modify-language-packs.png)

1. Choose **Modify**. The update starts.

### Modify language settings in Visual Studio

> Once you have the desired language packs installed, modify your Visual Studio settings to use a different language:

Reviewer: ```suggestion
Once you've installed the desired language packs, modify your Visual Studio settings to use a different language:
```

1. Open Visual Studio.

> 1. Choose **Continue without code**.

Reviewer: ```suggestion
1. On the start window, choose **Continue without code**.
```

---
title: Install localized IntelliSense files
description: Learn how to set up your development machine to use localized IntelliSense files.
author: mairaw
ms.author: mairaw
ms.date: 12/12/2019
---

> # How to install localized IntelliSense files for .NET Core projects

Reviewer 2: Should we make it clear this is only for Windows?

Author: This is even only for VS, right? Or would this work for VS Code too?

Reviewer 2: Cc @carlossanlop 

Not sure on VS Code part.

Reviewer 3: I'm not sure if this is specific for Visual Studio, because the zip file with the localized xmls needs to be extracted in this folder:

`C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\3.0.0\ref\netcoreapp3.0\`

But I only know how to change the IntelliSense language in Visual Studio. I don't know about VS Code or other IDEs.

Reviewer 3: My suggestion is to mention these instructions are specific for Visual Studio (which is Windows by default). I believe Maira already has that information mentioned in the document.

Thoughts, @vivmishra , @mairaw?

Author: I think so. We can investigate if this would work for VS Code too and edit the article if needed. 
This might work for VS for Mac too(?), but I currently don't have my Mac configured to test this. 
I'll just change the title to say Visual Studio as well and that should make it pretty clear for now. Introduction and prereqs already list VS.

> # How to install localized IntelliSense files for .NET Core projects

Reviewer 4: You could probably drop the "projects" part of the title. These intellisense files could really be used for anything by anything. The user is installing them for the .NET Core product itself, not for projects. Otherwise, this may lead someone to think that if they want to "install intellisense for their own project" they follow these same steps.

IntelliSense is a code-completion aid that is available in different integrated development environments (IDEs), such as Visual Studio. By default, when you're developing .NET Core projects, the SDK only includes the English version of the IntelliSense files. This article explains how to install the localized version of those files and modify the Visual Studio installation to use a different language.

## Prerequisites

- [.NET Core 3.0 SDK](https://dotnet.microsoft.com/download/dotnet-core) or later versions
- [Visual Studio 2019 16.3 or later versions](https://visualstudio.microsoft.com/downloads/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link&utm_content=download+vs2019)

## Download and install the localized IntelliSense files

> 1. Go to the [Download IntelliSense files](https://dotnet.microsoft.com/download/dotnet-core) page.

Reviewer 2: I am assuming that this link we will eventualy update to the new Intellisense page that is coming up.

Author: Yes 😄

Author: I've updated to what I think it will be the final URL. Looks right?

      | .NET Core       | *Microsoft.NETCore.App.Ref*        |
      | Windows Desktop | *Microsoft.WindowsDesktop.App.Ref* |
      | .NET Standard   | *NETStandard.Library.Ref*          |
   
   - Navigate to the version you want to install the localized IntelliSense for. For example, *3.1.0*.
   - Open the *ref* folder.
   - Open the moniker folder. For example, *netcoreapp3.1*.

   So, the full path that you'd navigate to would look similar to *C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Ref\3.1.0\ref\netcoreapp3.1*.

1. Create a subfolder inside the moniker folder you just opened. The name of the folder indicates which language you want to use. The following table specifies the different options:

| Language              | Folder name |
| --------------------- | ----------- |
| Chinese (simplified)  | *zh-hans*   |
| Chinese (traditional) | *zh-hant*   |
| French                | *fr*        |
| German                | *de*        |
| Italian               | *it*        |
| Japanese              | *ja*        |
| Korean                | *ko*        |
| Russian               | *ru*        |
| Spanish               | *es*        |

> 

Reviewer 5: Shouldn't Brazilian Portuguese (pt-br) also be included here?

Author: I use the same languages we had in https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/PackageLibs.targets#L59-L91 as @carlossanlop pointed out. Not sure if it would work for pt-br.

Reviewer 5: I tested this and it seems to work for pt-br too:
![pt-br_intelliSense](https://user-images.githubusercontent.com/18313964/70839294-e61fee00-1dc0-11ea-9dec-1aeddde422de.png)

1. Copy all the *.xml* files you extracted on step 3 to this new folder. If you're copying this to the .NET Standard path, then you can just copy the *netstandard.xml* file.

> ## Modify Visual Studio language

Author: I'll add some instructions here but once https://github.com/MicrosoftDocs/visualstudio-docs/issues/4491 gets fixed, I'll just point to the VS docs instead for the VS steps.
